.github/workflows/monitor_pipelines.sh

```diff
@@ -0,0 +1,137 @@
+#!/bin/bash
+
+set -e
+
+# Check if execution ID is provided
+if [ $# -ne 1 ]; then
+    echo "Usage: $0 <execution-id>"
+    exit 1
+fi
+
+EXECUTION_ID=$1
+MAIN_PIPELINE="koffir-workshop-test"
+
+echo "Checking execution $EXECUTION_ID of pipeline $MAIN_PIPELINE..."
+
+# Check the status of the main pipeline execution
+MAIN_STATUS=$(aws codepipeline get-pipeline-execution \
+    --pipeline-name "$MAIN_PIPELINE" \
+    --pipeline-execution-id "$EXECUTION_ID" \
+    --query 'pipelineExecution.status' \
+    --output text)
+
+echo "Main pipeline status: $MAIN_STATUS"
+
+# If the main pipeline is not successful, exit with error
+if [ "$MAIN_STATUS" != "InProgress" ]; then
+    echo "Main pipeline did not started. Exiting."
+    exit 1
+fi
+
+echo "Looking for Observability Workshop Pipeline..."
+sleep 30
+
+# Find the Observability Workshop Pipeline
+WORKSHOP_PIPELINE=$(aws codepipeline list-pipelines \
+    --query "pipelines[?starts_with(name, 'Observability-Workshop-Pipeline-')].name" \
+    --output text)
+
+if [ -z "$WORKSHOP_PIPELINE" ]; then
+    echo "No Observability Workshop Pipeline found. Exiting."
+    exit 1
+fi
+
+echo "Found pipeline: $WORKSHOP_PIPELINE"
+
+# Get the latest execution ID of the workshop pipeline
+WORKSHOP_EXECUTION_ID=$(aws codepipeline list-pipeline-executions \
+    --pipeline-name "$WORKSHOP_PIPELINE" \
+    --max-items 1 \
+    --query 'pipelineExecutionSummaries[0].pipelineExecutionId' \
+    --output text)
+
+echo "Monitoring execution $WORKSHOP_EXECUTION_ID of pipeline $WORKSHOP_PIPELINE..."
+
+# Monitor the workshop pipeline execution until completion
+
+WORKSHOP_STATUS=$(aws codepipeline get-pipeline-execution \
+    --pipeline-name "$WORKSHOP_PIPELINE" \
+    --pipeline-execution-id "$WORKSHOP_EXECUTION_ID" \
+    --query 'pipelineExecution.status' \
+    --output text)
+
+echo "Current status: $WORKSHOP_STATUS"
+
+if [ "$WORKSHOP_STATUS" == "Failed" ] || [ "$WORKSHOP_STATUS" == "Stopped" ]; then
+    echo "Pipeline execution failed or was stopped."
+    exit 1
+fi
+    
+    echo "Waiting for pipeline to complete..."
+    sleep 30
+done
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Parsing stopped here. Is this keyword correctly matched up?